edm4hep/CMakeLists.txt

```diff
@@ -7,6 +7,12 @@ PODIO_ADD_DATAMODEL_CORE_LIB(edm4hep "${headers}" "${sources}")
 add_library(EDM4HEP::edm4hep ALIAS edm4hep)
 set_target_properties(edm4hep PROPERTIES ALIAS_GLOBAL true)
 
+file(GLOB_RECURSE top_headers ${PROJECT_SOURCE_DIR}/include/*.h)
+target_sources(edm4hep PUBLIC FILE_SET headers TYPE HEADERS BASE_DIRS ${PROJECT_SOURCE_DIR}/include FILES ${top_headers})
+target_include_directories(edm4hep PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
```

Comment:
Reading from cmake docs:
> Target properties related to include directories are also modified by target_sources(FILE_SET) as follows:
>
>[INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INCLUDE_DIRECTORIES.html#prop_tgt:INCLUDE_DIRECTORIES)
>
>    If the TYPE is HEADERS, and the scope of the file set is PRIVATE or PUBLIC, all of the BASE_DIRS of the file set are wrapped in [$<BUILD_INTERFACE>](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:BUILD_INTERFACE) and appended to this property.
>
>[INTERFACE_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html#prop_tgt:INTERFACE_INCLUDE_DIRECTORIES)
>
>    If the TYPE is HEADERS, and the scope of the file set is INTERFACE or PUBLIC, all of the BASE_DIRS of the file set are wrapped in [$<BUILD_INTERFACE>](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:BUILD_INTERFACE) and appended to this property.

a `FILE_SET` updates the `INCLUDES_DIRECTORY` so manual setting with `target_include_directories` is redundant. These lines could be removed

Reply:
The `INSTALL_INTERFACE` line is not redundant in principle but it happens the public headers are installed to the include dir and that has the same effect as adding here the INSTALL_INTERFACE, so these lines can be removed.

Reply:
>  the public headers are installed to the include dir and that has the same effect as adding here the INSTALL_INTERFACE

I don't understand why the installation to the include dir has the same effect as adding the INSTALL_INTERFACE. Do you refer to PODIO_GENERATE_DATAMODEL (or the other podio macro) already setting the INSTALL_INTERFACE?

Reply:
Both `target_include_directories` and `install` populate `INTERFACE_INCLUDE_DIRECTORIES`:
From the [install documentation](https://cmake.org/cmake/help/latest/command/install.html)
```
INCLUDES DESTINATION

    This option specifies a list of directories which will be added to the [INTERFACE_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html#prop_tgt:INTERFACE_INCLUDE_DIRECTORIES) target property of the <targets> when exported by the [install(EXPORT)](https://cmake.org/cmake/help/latest/command/install.html#export) command. If a relative path is specified, it is treated as relative to the [$<INSTALL_PREFIX>](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:INSTALL_PREFIX).
```
So in this case we already have the install command populating `INTERFACE_INCLUDE_DIRECTORIES`, having also `target_include_directories` with the `INSTALL_INTERFACE` means we'll get the same path duplicated in `INTERFACE_INCLUDE_DIRECTORIES`

Reply:
Ahh, this is set from podio:
https://github.com/AIDASoft/podio/blob/master/cmake/podioMacros.cmake#L250